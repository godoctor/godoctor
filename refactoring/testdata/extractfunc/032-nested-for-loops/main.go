//<<<<<extract,10,3,20,3,Foo,pass
package main

import "fmt"

func main() {
	i := 1
	for k := 0; k < 3; k++ {
		fmt.Println("k =", k)
		for j := k; j < 5; j++ {
			fmt.Println("j =", j)
			fmt.Println("i =", i)
			if i%2 == 0 {
				fmt.Println("i is ", i)
				if i == 4 {
					break
				}
			}
			i++
		}
		fmt.Println("out of j loop")
	}
	fmt.Println("out of k loop")
}
