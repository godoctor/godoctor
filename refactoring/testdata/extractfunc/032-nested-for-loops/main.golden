//<<<<<extract,10,3,20,3,Foo,pass
package main

import "fmt"

func main() {
	i := 1
	for k := 0; k < 3; k++ {
		fmt.Println("k =", k)
		i = Foo(i, k)
		fmt.Println("out of j loop")
	}
	fmt.Println("out of k loop")
}
func Foo(i int, k int) int {
	for j := k; j < 5; j++ {
		fmt.Println("j =", j)
		fmt.Println("i =", i)
		if i%2 == 0 {
			fmt.Println("i is ", i)
			if i == 4 {
				break
			}
		}
		i++
	}
	return i
}
